pkg/controllers/repositoryserver/server.go

```diff
@@ -97,6 +97,32 @@ func (h *RepoServerHandler) getServerDetails(ctx context.Context) (string, strin
 }
 
 func (h *RepoServerHandler) checkServerStatus(ctx context.Context, serverAddress, username, password string) error {
+	fingerprint, err := kopia.ExtractFingerprintFromCertSecret(ctx, h.KubeCli, h.RepositoryServerSecrets.serverTLS.Name, h.RepositoryServer.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "Failed to extract fingerprint from Kopia API server certificate secret data")
+	}
+	cmd := command.ServerStatus(
+		command.ServerStatusCommandArgs{
+			CommandArgs: &command.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: command.DefaultConfigFilePath,
+				LogDirectory:   command.DefaultLogDirectory,
+			},
+			ServerAddress:  serverAddress,
+			ServerUsername: username,
+			ServerPassword: password,
+			Fingerprint:    fingerprint,
+		})
+	stdout, stderr, exErr := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+	format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+	format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+	if exErr != nil {
+		return exErr
+	}
+	return nil
+}
+
+func (h *RepoServerHandler) checkServerStatusTillCommandSucceed(ctx context.Context, serverAddress, username, password string) error {
```

Comment:
If I look at this function name and have to guess what does this function does. I would guess it accepts a command and waits for that command to run successfully. And we are not doing that. 
Instead we are actually waiting for the server to be ready (if I am not mistaken), so let's name is the same `waitForServerReady`.

Reply:
Done !